src/Subscriber/CategoryPageSubscriber.php

```diff
@@ -12,26 +12,32 @@
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use function Omikron\FactFinder\Shopware6\Internal\Utils\safeGetByName;
 
-class CategoryView implements EventSubscriberInterface
+class CategoryPageSubscriber implements EventSubscriberInterface
 {
+    private const ADD_PARAMETER_VALUE_PATTERN = '%s=%s';
+
     private AbstractCategoryRoute $cmsPageRoute;
 
     private CategoryPath $categoryPath;
 
     private string $fieldName;
 
-    private array $addParams;
+    private array $categoryPageAddParams;
+
+    private array $configurationAddParams;
 
     public function __construct(
         AbstractCategoryRoute $cmsPageRoute,
         CategoryPath $categoryPath,
         string $categoryPathFieldName,
-        array $initialNavigationParams = []
+        array $categoryPageAddParams = [],
+        array $configurationAddParams = []
     ) {
-        $this->cmsPageRoute = $cmsPageRoute;
-        $this->categoryPath = $categoryPath;
-        $this->fieldName    = $categoryPathFieldName;
-        $this->addParams    = $initialNavigationParams;
+        $this->cmsPageRoute           = $cmsPageRoute;
+        $this->categoryPath           = $categoryPath;
+        $this->fieldName              = $categoryPathFieldName;
+        $this->categoryPageAddParams  = $categoryPageAddParams;
+        $this->configurationAddParams = $configurationAddParams;
```

Comment:
This could be done easier. You don't need to require the $configurationAddParams as they will be available in the $event object.

``
$baseAddParams = explode(',', safeGetByName($event->getPage()->getExtension('factfinder')->get('communication'))('add-params'));``

safeGetByName is used only for safety reason. It's possible that 

``
$event->getPage()->getExtension('factfinder')->get('communication)'
``
will return array where `add-params` key does not exist.
So after doing that we have array of expressions "key=value" defined in ConfigurationSubscriber

Reply:
fixed